File: NetFramework/XHtmlEditor/VB/RichTextEditor/AppMenuTabs/Items/AppMenuTabButton.vb
Imports System.Drawing
Imports System.Windows.Forms

Partial Public Class AppMenuTabButton
    Inherits UserControl

    Private _defaultBackColor As Color = Color.FromArgb(212, 212, 212)
    Private _hotBackColor As Color = Color.DarkGray
    Private _hotPressedBackColor As Color = Color.Gray

    Public Sub New()
        InitializeComponent()
        BackColor = _defaultBackColor
    End Sub

    Public Property ItemImage As Image
        Get
            Return pbImage.Image
        End Get

        Set(ByVal value As Image)
            pbImage.Image = value
        End Set
    End Property

    Public Property ItemText As String
        Get
            Return lblText.Text
        End Get

        Set(ByVal value As String)
            lblText.Text = value
        End Set
    End Property

    Private Sub MenuItem_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs) Handles pbImage.MouseDown, lblText.MouseDown, MyBase.MouseDown
        BackColor = _hotPressedBackColor
    End Sub

    Private Sub MenuItem_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs) Handles pbImage.MouseUp, lblText.MouseUp, MyBase.MouseUp
        BackColor = _hotBackColor
    End Sub

    Private Sub MenuItem_MouseLeave(ByVal sender As Object, ByVal e As EventArgs) Handles pbImage.MouseLeave, lblText.MouseLeave, MyBase.MouseLeave
        BackColor = _defaultBackColor
    End Sub

    Private Sub MenuItem_MouseEnter(ByVal sender As Object, ByVal e As EventArgs) Handles pbImage.MouseEnter, lblText.MouseEnter, MyBase.MouseEnter
        BackColor = _hotBackColor
    End Sub

    Private Sub MenuItem_Click(ByVal sender As Object, ByVal e As EventArgs) Handles pbImage.Click, lblText.Click
        OnClick(e)
    End Sub

End Class

File: NetFramework/XHtmlEditor/CS/RichTextEditor/AppMenuTabs/Items/AppMenuTabButton.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace RichTextEditor.AppMenuTabs.Items
{
    public partial class AppMenuTabButton : UserControl
    {
        #region private fields
        private Color _defaultBackColor = Color.FromArgb(212, 212, 212);
        private Color _hotBackColor = Color.DarkGray;
        private Color _hotPressedBackColor = Color.Gray;
        #endregion

        #region ctor
        public AppMenuTabButton()
        {
            InitializeComponent();
            BackColor = _defaultBackColor;
        }
        #endregion

        #region properties
        public Image ItemImage
        {
            get { return pbImage.Image; }
            set { pbImage.Image = value; }
        }

        public string ItemText
        {
            get { return lblText.Text; }
            set { lblText.Text = value; }
        }
        #endregion

        #region event handlers
        private void MenuItem_MouseDown(object sender, MouseEventArgs e)
        {
            BackColor = _hotPressedBackColor;
        }

        private void MenuItem_MouseUp(object sender, MouseEventArgs e)
        {
            BackColor = _hotBackColor;
        }

        private void MenuItem_MouseLeave(object sender, EventArgs e)
        {
            BackColor = _defaultBackColor;
        }

        private void MenuItem_MouseEnter(object sender, EventArgs e)
        {
            BackColor = _hotBackColor;
        }

        private void MenuItem_Click(object sender, EventArgs e)
        {
            OnClick(e);
        }
        #endregion

    }
}
